### PR TITLE
update snapshot restore e2e to check data consistency

### PR DESCRIPTION
**Pull request by yati1998**
# Updates snapshot restore e2e to check data consistency
Currently, in rbd snapshot restore and volume clone E2E we are not checking any data consistency after doing snapshot restore or volume clone. Hence, this PR writes the data in the PVC and checks the md5sum of the file and verify it with the snapshot or cloned PVC.

Fixes: #1291

### CODE REVIEW

File: e2e/utils.go [hunk 1]
```diff
@@ -351,6 +351,28 @@ func writeDataInPod(app *v1.Pod, f *framework.Framework) error {
 	return err
 }
 
+//calculate md5sum
+func calMd5sum(app *v1.Pod, f *framework.Framework) (string, error) {
```
Comment (nixpanic) [inline e2e/utils.go:355]: @Yuggupta27 introduces `calculateSHA512sum()` for similar tests, please check with him and use a single checksum function

Comment (Yuggupta27) [reply e2e/utils.go:355]: Yes please, You can find it here https://github.com/ceph/ceph-csi/pull/1464/files#diff-c62628027491c2b96afe9b96aa706363R221

**Review by Yuggupta27 — CHANGES_REQUESTED** [review]
Can you please update commit message as per format mentioned here https://github.com/ceph/ceph-csi/blob/master/docs/development-guide.md#commit-messages

**Comment by yati1998** [comment]
Updated with new PR:  https://github.com/ceph/ceph-csi/pull/1557

**Pull request closed without merging**